design-proposals/nic-hotplug/nic-hotplug.md

```diff
@@ -0,0 +1,407 @@
+# Overview
+This KubeVirt design proposal discusses possible implementations for adding -
+and removing - network interfaces from running Virtual Machines, without
+requiring a restart.
+
+## Motivation
+Hot-plug / hot-unplug (add / remove) nics to running VMs is an industry
+standard available in multiple platforms, allowing the dynamic attachment of L2
+networks. This is useful when the workload (VM) cannot tolerate a restart when
+attaching / removing networks, of for scenarios where, for instance, the
+workload is created prior to the network.
+
+This would help bridge the gap between KubeVirt's feature set and what the user
+base expects from it.
+
+## Goals
```

Comment:
Could you please add 'container/Pod' in addition to VMs because multus is CNI ('container' network interface) plugin and multus need to support container...

Reply:
This is a document intended for KubeVirt, but since I'm using it also to describe plenty of behavioral changes for multus, it's only fair that I add pods as goals. 

Will add these, along with a short disclaimer that "adding / removing network interfaces to pods **is a requirement** for adding them to VMs".

Reply:
Done.